src/libs/antares/study/parts/thermal/cluster.cpp

```diff
@@ -434,6 +474,72 @@ void Data::ThermalCluster::calculationOfSpinning()
     }
 }
 
+void Data::ThermalCluster::calculationOfMarketBidPerHourAndMarginalCostPerHour()
+{
+    if (costgeneration == Data::setManually || !prepro || efficiency <= 0.0)
```

Comment:
- Do not test `prepro` here
- `efficiency <= 0.0` should be an error, reported earlier